trace.go

```diff
@@ -59,31 +59,50 @@ func Unwrap(err error) error {
 	return err
 }
 
+// UserMessage returns user-friendly part of the error
+func UserMessage(err error) string {
+	if err == nil {
+		return ""
+	}
+	if wrap, ok := err.(Error); ok {
+		return wrap.UserMessage()
+	}
+	return err.Error()
+}
+
+// DebugReport returns debug report with all known information
+// about the error including stack trace if it was captured
+func DebugReport(err error) string {
+	if err == nil {
+		return ""
+	}
+	if wrap, ok := err.(Error); ok {
+		return wrap.DebugReport()
+	}
+	return err.Error()
+}
+
 func wrap(err error, depth int, args ...interface{}) Error {
```

Comment:
I think you can simplify the use of this function by removing `depth` argument, as I've only seen a single use for it with `depth=2`. If it only exists to augment the result of `newTrace` with an user message (and avoid over-wrapping) - maybe Errorf should be using it too?